.github/workflows/build.yml

```diff
@@ -0,0 +1,170 @@
+name: Build Artifacts
+
+on:
+  workflow_call:
+    inputs:
+      version:
+        required: true
+        type: string
+      unix:
+        default: true
+        type: boolean
+      windows32:
+        default: true
+        type: boolean
+
+  workflow_dispatch:
+    inputs:
+      version:
+        description: |
+          VERSION: yyyy.mm.dd[.rev] or rev
+        required: true
+        type: string
+      unix:
+        description: youtube-dl, youtube-dl.tar.gz
+        default: true
+        type: boolean
+      windows32:
+        description: youtube-dl.exe
+        default: true
+        type: boolean
+
+permissions:
+  contents: read
+
+jobs:
+  unix:
+    if: inputs.unix
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Needed for changelog
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.9"
+
+      - name: Install Requirements
+        run: |
+          sudo apt -y install zip pandoc man sed
+
+      - name: Prepare
+        run: |
+          python devscripts/update_version.py "${{ inputs.version }}"
+          python devscripts/changelog.py --update
+          python devscripts/make_lazy_extractors.py youtube_dl/extractor/lazy_extractors.py
+
+      - name: Build Unix platform-independent binary
+        run: |
+          make all tar
+
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: build-bin-${{ github.job }}
+          path: |
+            youtube-dl
+            youtube-dl.tar.gz
+          compression-level: 0
+
+  windows32:
+    if: inputs.windows32
+    runs-on: windows-2022
+    env:
+      PYCRYPTO: pycrypto-2.6.1-cp34-none-win32
+      # Temporary workaround for Python 3.4/5 failures - May 2024
+      PIP_TRUSTED_HOST: "pypi.python.org pypi.org files.pythonhosted.org"
```

Comment:
Coped over, but no idea what this means

Reply:
The TLS parameters for the listed hosts have been upgraded so that you can't connect with an old _pip_ -- [except with this work-around](https://github.com/ytdl-org/youtube-dl/commit/a08f2b7e4567cdc50c0614ee0a4ffdff49b8b6e6), essentially a `--no-check-certificate` for those hosts.

Security measures as denial-of-service: a very relevant issue this week.